internal/checker/utilities.go

```diff
@@ -1218,8 +1218,84 @@ func isValidNumberString(s string, roundTripOnly bool) bool {
 	return !n.IsNaN() && !n.IsInf() && (!roundTripOnly || n.String() == s)
 }
 
-func isValidBigIntString(s string, roundTripOnly bool) bool {
-	return false // !!!
+func isValidBigIntString(s string, roundtrip bool) bool {
+	if s == "" {
+		return false
+	}
+
+	// Create a scanner and set the text to the input string with an 'n' suffix
+	scan := scanner.NewScanner("", scanner.ScriptTargetESNext, false /*skipTrivia*/)
+	text := s + "n"
+	scan.SetText(text)
+	
+	// Scan the first token
+	token := scan.Scan()
+	
+	// Check for a negative sign
+	negative := token == ast.KindMinusToken
+	if negative {
+		token = scan.Scan()
+	}
+	
+	tokenFlags := scan.GetTokenFlags()
+	
+	// Validate that:
+	// 1. We scanned a BigIntLiteral token
+	// 2. The scanner consumed the entire input (plus the 'n' we added)
+	// A correct BigInt literal does not contain separators
+	if token != ast.KindBigIntLiteral || 
+	   scan.GetTokenEnd() != (len(s) + 1) || 
+	   (tokenFlags & scanner.TokenFlagsContainsSeparator) != 0 {
+		return false
+	}
+	
+	// If roundtrip checking is required, verify the string representation matches the original
+	if roundtrip {
```

Comment:
I'd just write `if !roundtrip { return true }` instead; the original code bailed early.

Reply:
yep, totally. will fix.